File: EventimFrontend/EventimFrontend/src/app/Models/event.ts
import {Ticket} from "./ticket";

export interface Event {
  id: number;
  eventName: string;
  date?: string; // Datum als Zeichenkette für einfache Darstellung
  ticketAmount?: number;
  tickets?: Ticket[]; // 1:N-Beziehung zu Tickets
}

File: EventimBackend/src/main/java/com/example/eventimbackend/entities/Event.java
package com.example.eventimbackend.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Entity
@Data
public class Event {

    @Id
    @GeneratedValue
    private Long id;

    private String eventName;

    private LocalDate date;

    private Integer ticketAmount;

    @OneToMany(mappedBy = "event")
    @JsonIgnoreProperties("event")
    private List<Ticket> tickets;

}
